File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.py
def solution(targets):
    # s, s가 같으면 e 순으로 정렬하기
    targets.sort()

    # 초기에는 무조건 미사일 하나 소비함
    answer = 1
    # 현재 미사일이 있는 좌표 초기화
    idx = targets[0][1] - 1

    for target in targets:
        if target[0] - 1 < idx:
            idx = min(idx, target[1] - 1)
        else:
            answer += 1
            idx = target[1] - 1

    return answer

File: 프로그래머스/2/181188. 요격 시스템/요격 시스템.java
import java.util.*;


class Solution {
    public int solution(int[][] targets) {
        // targets의 끝나는 e에 따라서 오름차순 정렬
        Arrays.sort(targets, new Comparator<int[]>(){
            @Override 
            public int compare(int[] o1, int[] o2){
                if(o1[1] == o2[1]){
                    return o1[0] - o2[0];
                }
                return o1[1] - o2[1];
            }
        });
        int answer = 0;
        // 이차원 어레이를 출력하는 코드
        // System.out.println(Arrays.deepToString(targets));
        int ce = 0;
        
        for(int[] info:targets){
            if(info[0] >= ce){
                answer += 1;
                ce = info[1];
            }
        }
        return answer;
    }
    
    
}